### PR TITLE
shapelib: update from upstream: general cleanup; no functional changes

### PR DESCRIPTION
**Pull request by schwehr**
Upstream at https://github.com/OSGeo/shapelib/tree/2025ce22196f100a773e2d0bbdc86cfd2ad0ad18

General cleanup.  No functional changes.

- Use stdbool true/false for internal variables
- Localize variables
- Combine variable definition and initialization when possible
- Add const when possible
- Remove or comment out most unused variables
- Remove some extra empty space
- Add some missing headers
- Sort standard headers
- Move static variable to after includes
- Remove RCS/CVS/SVN Id/Revision tags except for SHP_CVSID
- Remove some "goto error;" from shpopen.c

### CODE REVIEW

File: gdal/ogr/ogrsf_frmts/shape/shapefil.h [hunk 1]
```diff
@@ -10,7 +10,7 @@
  *
  ******************************************************************************
  * Copyright (c) 1999, Frank Warmerdam
- * Copyright (c) 2012-2016, Even Rouault <even dot rouault at spatialys.com>
+ * Copyright (c) 2012-2016, Even Rouault <even dot rouault at mines-paris dot org>
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/shape/shapefil.h:13]: should be reverted here / updated in shapelib

File: gdal/ogr/ogrsf_frmts/shape/sbnsearch.c [hunk 2]
```diff
@@ -1,12 +1,13 @@
+#define sanity_checks 1
+#define DEBUG_IO 1
 /******************************************************************************
- * $Id$
  *
  * Project:  Shapelib
  * Purpose:  Implementation of search in ESRI SBN spatial index.
- * Author:   Even Rouault, even dot rouault at spatialys.com
+ * Author:   Even Rouault, even dot rouault at mines dash paris dot org
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/shape/sbnsearch.c:7]: should be reverted here / updated in shapelib

File: gdal/ogr/ogrsf_frmts/shape/sbnsearch.c [hunk 3]
```diff
@@ -1,12 +1,13 @@
+#define sanity_checks 1
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/shape/sbnsearch.c:1]: should be reverted

File: gdal/ogr/ogrsf_frmts/shape/sbnsearch.c [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
+#define DEBUG_IO 1
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/shape/sbnsearch.c:2]: should be reverted

File: gdal/ogr/ogrsf_frmts/shape/sbnsearch.c [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
- * $Id$
  *
  * Project:  Shapelib
  * Purpose:  Implementation of search in ESRI SBN spatial index.
- * Author:   Even Rouault, even dot rouault at spatialys.com
+ * Author:   Even Rouault, even dot rouault at mines dash paris dot org
  *
  ******************************************************************************
- * Copyright (c) 2012-2014, Even Rouault <even dot rouault at spatialys.com>
+ * Copyright (c) 2012-2014, Even Rouault <even dot rouault at mines-paris dot org>
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/shape/sbnsearch.c:10]: should be reverted here / updated in shapelib

File: gdal/ogr/ogrsf_frmts/shape/shptree.c [hunk 6]
```diff
@@ -7,7 +7,7 @@
  *
  ******************************************************************************
  * Copyright (c) 1999, Frank Warmerdam
- * Copyright (c) 2012, Even Rouault <even dot rouault at spatialys.com>
+ * Copyright (c) 2012, Even Rouault <even dot rouault at mines-paris dot org>
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/shape/shptree.c:10]: should be reverted here / updated in shapelib

File: gdal/ogr/ogrsf_frmts/shape/sbnsearch.c (hunk 3 above)
Comment (schwehr) [reply gdal/ogr/ogrsf_frmts/shape/sbnsearch.c:1]: Thanks for the catch.

File: gdal/ogr/ogrsf_frmts/shape/sbnsearch.c (hunk 2 above)
Comment (schwehr) [reply gdal/ogr/ogrsf_frmts/shape/sbnsearch.c:7]: https://github.com/OSGeo/shapelib/commit/3e59b35009f81538c2f3014a20c7abfc7a964ba3

File: gdal/ogr/ogrsf_frmts/shape/sbnsearch.c (hunk 3 above)
Comment (schwehr) [reply gdal/ogr/ogrsf_frmts/shape/sbnsearch.c:1]: https://github.com/OSGeo/shapelib/commit/d9aed4b3254051cea1fe23a28ef5b873e0d926bb

**Comment by schwehr** [comment]
@rouault Thanks for the review!

**Pull request merged**